modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/service/impl/MBCategoryLocalServiceImpl.java

```diff
@@ -959,7 +959,9 @@ private String _getUniqueFriendlyURL(
 			groupId, friendlyURL);
 
 		for (int i = 1; mbCategory != null; i++) {
-			friendlyURL = name + StringPool.DASH + i;
+			friendlyURL =
+				_friendlyURLNormalizer.normalizeWithEncodingPeriodsAndSlashes(
+					name + StringPool.DASH + i);
```

Comment:
Would it be possible to move the call to `friendlyURLNormalizer` outside of the loop. If I've understood the code correctly, as the `-<n>` suffix will always normalize to itself, this should be equivalent:
```
String normalizedName = _friendlyURLNormalizer.normalizeWithEncodingPeriodsAndSlashes(name);

for (int i = 1; mbCategory != null; i++) {
    friendlyURL = normalizedName + StringPool.DASH + i;
    ...
}
```
This way, we avoid calling the normalizer on each iteration.

Reply:
Also, can you add a test that verifies that something already normalized normalizes to itself? e.g. `foo-1` -> `foo-1`.

Reply:
Maybe I can move the call with the dash, something like :  

1-_friendlyURLNormalizer.normalizeWithEncodingPeriodsAndSlashes(name + StringPool.DASH)

or:

2- see if we already have a dash in the normalized friendly url to avoid have double dash.

Because the problema is , if we normalized the url and for some reasons this url ends with dash, the logic of the normalization process is to do not have two followed dash as we are already doing for the other normalize method.

So, I think that the the second option is my favorite. I will go with that and also I will add another test case to test that.